### PR TITLE
Prefactor: Extract logger interface

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The point here is to extract an interface from the Logger type in order to be able to instantiate mocks and further test the types wrapping around loggers.

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should be sufficient.

Manual testing by building and deploying a local image with the changes and check that the container logs have the expected form.

### CODE REVIEW

**Comment by rhybrillou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5853** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5853" target="_blank">Graphite</a>  👈
    * **PR #5854** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5854" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/5853?utm_source=stack-comment).

File: pkg/logging/logger.go [hunk 1]
```diff
@@ -5,14 +5,38 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
-// Logger wraps a zap.SugaredLogger.
-type Logger struct {
+// Logger is the interface exposed for logging purposes
```
Comment (dhaus67) [inline pkg/logging/logger.go:8]:
```suggestion
// Logger is the interface exposed for logging purposes.
```

File: pkg/grpc/logging/logging.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 
 // UnaryServerInterceptor returns a grpc.UnaryServerInterceptor that logs incoming requests
 // and associated tags to "logger".
-func UnaryServerInterceptor(logger *logging.Logger) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(logger *logging.LoggerImpl) grpc.UnaryServerInterceptor {
```
Comment (dhaus67) [inline pkg/grpc/logging/logging.go:13]: Similar to that, we can avoid using `logging.LoggerImpl` here by extending the interface to have a `SugardedLogger()` method which will return the sugared logger. This way we avoid exposing the interface's implementation and only have a single "thing" people may reference.

File: pkg/logging/internal/internal_test.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ func TestCurrentModule(t *testing.T) {
 }
 
 func TestLoggerForModule(t *testing.T) {
-	assert.Equal(t, "pkg/logging/internal", logging.LoggerForModule().Module().Name())
+	assert.Equal(t, "pkg/logging/internal", logging.CurrentModule().Logger().Module().Name())
```
Comment (janisz) [inline pkg/logging/internal/internal_test.go:15]: I think this change is not necessary

**Review by janisz — APPROVED** [review]

**Pull request merged**